### ISSUE TITLE
Ability to close files 

### ISSUE THREAD

**Issue by hibit-2709** [danmunn/redmine_dmsf#1278]
Are you planned to introduce the ability to 'close' files?
Authorized users could close (hide) files. Other eligible people could continue to see them.
Permissions would be set in module settings (not project or file)

**Comment by picman** [comment]
You can hide files using permissions defined on folder level.

**Comment by hibit-2709** [comment]
But it blocks all files from the folder.
I'm looking for a solution to the problem of ending the current ness of a given document.
Builders should continue to see it and production workers should not.

**Comment by picman** [comment]
Then you can create a folder invisible for production staff and move closed file into this folder.
A "close" function doesn't make sense for me in DMS. In a standard file system you also cannot close a file, but you can move it to a folder which is visible for certain users only.

**Comment by hibit-2709** [comment]
Yes, you're right. However, moving the document requires creating a folder for the archives. We have several thousand projects. Moving a file itself is time consuming, while clicking on 'Closed' seems to be easier just like locking it.
Thanks for the attention. I'm looking for another solution.

**Comment by hibit-2709** [comment]
I noticed a regularity. If I move a file as an administrator, it takes me x seconds. However, when I do it as a regular user, the time is x * 3 (or even 5) seconds.

**Comment by picman** [comment]
The reason is very probably your large DMS structure with a lot of permissions defined. In case of an ordinary user I have to carefully check their permissions, in case of administrator I haven't to check anything.
Thus I don't mean that the code cannot be optimized. It would mean to analyze SQL queries run during a document movement and find a way how to optimize them.
